components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent') -- Note: This can be overwritten
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local WidgetInjector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+-- local LpdbInjector = Lua.import('Module...', {requireDevIfEnabled = true})
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+
+local TODAY = os.date('%Y-%m-%d')
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+	'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
+
+local specialDataTypes = Table.map({
+	'LASTVS', 'GROUPSCORE', 'LASTVSSCORE'
+}, function(value, name) return name, value end)
+
+local prizeData = {
+	[prizeTypes.USD] = {
+		row = 'usdprize',
+		rowParse = function (prizePool, input, context, index)
+			prizePool.hasUSD = true
+			return prizePool:_parseInteger(input)
+		end
```

Comment:
This function is not pure. One expects it to only handle parsing but it does more than that.

Reply:
Remove from here, but still not entirely happy with the new location of it neither.

Reply:
I think it's fine now. Could be better, but acceptable